concordium-consensus/src/Concordium/GlobalState/Persistent/TreeState.hs

```diff
@@ -901,9 +933,14 @@ instance
         finTrans WithMetadata{wmdData = NormalTransaction tr, ..} = do
             let nonce = transactionNonce tr
                 sender = accountAddressEmbed (transactionSender tr)
-            anft <- use (skovPersistentData . transactionTable . ttNonFinalizedTransactions . at' sender . non emptyANFT)
-            if anft ^. anftNextNonce == nonce
+            senderMinNonce <- do
+                st <- blockState =<< use (skovPersistentData . lastFinalized)
+                macct <- getAccount st (aaeAddress sender)
+                fromMaybe minNonce <$> mapM (getAccountNonce . snd) macct
+            nextNonce <- use (skovPersistentData . transactionTable . ttNonFinalizedTransactions . at' sender . non (emptyANFTWithNonce senderMinNonce) . anftNextNonce)
```

Comment:
I think this is a very bad idea, to look up each account during transaction finalization. That's not cheap.